README.md

```diff
@@ -0,0 +1,713 @@
+
+* Include joiners
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("Tést", {
+        ignoreJoiners: true
+    });
+    // regexp =>
+    // /T[\u00ad|\u200b|\u200c|\u200d]?(\u00E9|e\u0301)[\u00ad|\u200b|\u200c|\u200d]?s[\u00ad|\u200b|\u200c|\u200d]?t/gu
+    ```
+
+  * When `true`, regular expression matches for soft hyphens (`\u00ad`), zero width space (`\u200b`), zero width non-joiner (`\u200c`) and zero width joiners (`\u200d`) are included between each character.
+
+* Regular expression flags
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("Tést", {
+        flags: "giu"
+    });
+    // regexp => /T(\u00E9|e\u0301)st/giu
+    ```
+
+* Each callback
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("a tést", {
+        each: function(character, result, data, index) {
+            // Replace space with regular expression whitespace
+            // make sure to double escape the RegExp special characters "\\s"
+            return character === " " ? "\\s+" : result;
+        }
+    });
+    // regexp => /a\s+t(\u00E9|e\u0301)st/gu
+    ```
+
+* Done callback
+
+    ```js
+    var diacritics = require("diacritics-transliterator");
+    var regexp = diacritics.createRegExp("tést", {
+        done: function(array, joiner) {
+            // match whole words only
+            // make sure to double escape the RegExp special characters "\\b"
+            return "\\b" + array.join(joiner) + "\\b";
+        }
+    });
+    // regexp => /\bt(\u00E9|e\u0301)st\b/gu
+    ```
+
+If no diacritics are found, then the regular expression will include the original text.
+
+#### __Important Notes__
+
+* This method is not meant to be used on HTML, and will escape any regular expression-like characters (e.g. `^(test)$` string is converted into the following string `\^\(test\)\$`).
+* If you want to include the start `^` and end `$` anchors, use the `done` callback in a manner similar to the example above.
+* Any regular expression special characters added within the callback functions must be double escaped (e.g. `\S` becomes `\\S`) because the regular expression is created from a string.
+* Adding an escaped unicode character within the callbacks must not be double escaped (e.g. `\u00a0` contains a single backslash). If you have included any double escaped unicode as found in the database, use the [`diacritics.formatUnicode` method](#diacriticsformatunicodestring) to process it.
+
+### diacritics.replacePlaceholder(string [, options])
+
+Replaces placeholder(s) within the string with the targeted diacritic values:
+- The `string` parameter (type: `string`)  contains text and/or HTML with a diacritic placeholder value(s) to be replaced.
```

Comment:
I've included a placeholder option... you can modify it however you wish. Look at [this test](https://github.com/diacritics/node-diacritics-transliterator/commit/85e8cb6047ac371e7b50a71d66cb090aaa3abcf7#diff-65a9616c82f38fb3b6e9e6373a914566R537) which sets the diacritic to use `<!== values:{data} ==>`